concordium-sdk/src/test/testresources/smartcontractschema/lib.rs

```diff
@@ -0,0 +1,242 @@
+}
+
+/// Takes ListParam as parameter for ensuring correct serialization of ListParam
+#[receive(
+    contract = "java_sdk_schema_unit_test",
+    name = "list_param_test",
+    parameter = "ListParam",
+    error = "Error",
+    mutable
+)]
+fn list_param_test<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &mut impl HasHost<State, StateApiType = S>,
+) -> Result<(), Error> {
+    Ok(())
+}
+
+/// Takes AddressContainer as parameter for ensuring correct serialization of Address
+#[receive(
+    contract = "java_sdk_schema_unit_test",
+    name = "abstract_address_container_test",
+    parameter = "AbstractAddressContainer",
+    error = "Error",
+    mutable
+)]
+fn abstract_address_container_test<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &mut impl HasHost<State, StateApiType = S>,
+) -> Result<(), Error> {
+    Ok(())
+}
+
+/// Takes Address as parameter for ensuring correct serialization of Address when passed directly
+#[receive(
+    contract = "java_sdk_schema_unit_test",
+    name = "abstract_address_test",
+    parameter = "Address",
+    error = "Error",
+    mutable
+)]
+fn abstract_address_test<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &mut impl HasHost<State, StateApiType = S>,
+) -> Result<(), Error> {
+    Ok(())
+}
+
+/// Takes a ContractAddressContainer as parameter for ensuring correct serialization of ContractAddress
+#[receive(
+    contract = "java_sdk_schema_unit_test",
+    name = "contract_address_container_test",
+    parameter = "ContractAddressContainer",
+    error = "Error",
+    mutable
+)]
+fn contract_address_container_test<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &mut impl HasHost<State, StateApiType = S>,
+) -> Result<(), Error> {
+    Ok(())
+}
+/// Takes Address as parameter for ensuring correct serialization of Address when passed directly
+#[receive(
+    contract = "java_sdk_schema_unit_test",
+    name = "contract_address_test",
+    parameter = "ContractAddress",
+    error = "Error",
+    mutable
+)]
+fn contract_address_test<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &mut impl HasHost<State, StateApiType = S>,
+) -> Result<(), Error> {
+    Ok(())
+}
+
+/// Takes an AccountAddressContainer as parameter for ensuring correct serialization of AccountAddress
+#[receive(
+    contract = "java_sdk_schema_unit_test",
+    name = "account_address_container_test",
+    parameter = "AccountAddressContainer",
+    error = "Error",
+    mutable
+)]
+fn account_address_container_test<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &mut impl HasHost<State, StateApiType = S>,
+) -> Result<(), Error> {
+    Ok(())
+}
+
+/// Takes Address as parameter for ensuring correct serialization of Address when passed directly
+#[receive(
+    contract = "java_sdk_schema_unit_test",
+    name = "account_address_test",
+    parameter = "AccountAddress",
+    error = "Error",
+    mutable
```

Comment:
why mutable?

Reply:
(Question also goes for the other receive functions here)

Reply:
As an answer to all of the above. Most of this was generated by `cargo concordium init`. I've documented what is and isn't used